sos/component.py

```diff
@@ -192,7 +192,7 @@ def apply_options_from_cmdline(self, opts):
         for opt, val in codict.items():
             if opt not in cmdopts.arg_defaults.keys():
                 continue
-            if val is not None and val != opts.arg_defaults[opt]:
+            if val and val != opts.arg_defaults[opt]:
```

Comment:
This was recently changed in c2ddb50 to fix the `log-size=0` issue, #2334. Changing this back will cause a regression.